### ISSUE TITLE
Raise classic async state machines (runtime-async=off) to async/await

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#948]
## Pattern

Raise the **classic async state-machine lowering** back to `async`/`await`. With `runtime-async=off` (the dominant real-world mode, and everything not on bleeding-edge runtime-async), an async method lowers to an `AsyncTaskMethodBuilder` kickoff + a `<Method>d__N` state-machine struct whose `MoveNext` carries the await logic. The decompiler does not raise this today — the kickoff renders as raw builder setup and the `MoveNext` stays goto-dispatched.

This is invisible to the CoreLib corpus (which ships `runtime-async=on`, raised by `AwaitRecoveryPass`), so it's surfaced via the multi-mode fixture matrix — the async axis, `src/ILInspector.Decompiler.Fixtures.ClassicAsync` (see `tools/DecompilerHarness/README.md`, "Multi-mode fixture matrix", and `docs/decompiler-quality.md`).

## Measurement (the tracked signal)

```bash
dotnet build src/ILInspector.Decompiler.Fixtures.ClassicAsync -c Release
dotnet run --project tools/DecompilerHarness -c Release -- --library-report \
  artifacts/bin/ILInspector.Decompiler.Fixtures.ClassicAsync/release/ILInspector.Decompiler.Fixtures.ClassicAsync.dll
```

Baseline: **21 methods, 0/21 raised.**

- `structuring: conditional-branch` × 7 — the `<…>d__N::MoveNext` state machines (the goto state dispatch the structurer can't raise).
- `fidelity: (typed)` × 14 — the 7 kickoffs + 7 `SetStateMachine` boilerplate methods.

Done = these buckets reach 0 (the overlay reports 21/21 Full, round-tripping `runtime-async=off`).

## Hits (the owner's test set)

7 source methods in `AsyncFixtures`, each → kickoff + `MoveNext` + `SetStateMachine` (21 methods):

- `AwaitValue` — single await, value-returning (`await a + b`)
- `AwaitVoid` — single await, `Task`
- `TwoSequentialAwaits` — two awaits in sequence
- `AwaitValueTask` — `ValueTask<T>`
- `AwaitInLoop` — await inside a `foreach`
- `AwaitInTryFinally` — await inside `try/finally`
- `AwaitConditional` — await in a branch

## The owed raise

- **Kickoff** (spiked on `feature/async-reconstruction` as `AsyncShapes`): returns Task/ValueTask, hands off to `builder.Start` over a `<M>d__N` struct, maps hoisted fields → parameters. Mark `async`, drop the builder scaffolding.
- **MoveNext**: strip the `try/catch(SetException)`/`SetResult` envelope and the `if (state==…) goto resume` dispatch; fold each await triple (`X.GetAwaiter()` … `AwaitUnsafeOnCompleted` … `awaiter.GetResult()`) into `await X` at the `GetResult` site; `builder.SetResult(v)` → `return v`. Reuse the existing `AwaitExpression` node and the iterator import-seam.

Structural analog: `IteratorReconstructionPass` (and `ReducibleIteratorReconstruction` for the loop/branch shapes). `AwaitRecoveryPass` is the runtime-async sibling.

## Suggested slices (incremental, one agent owns all)

1. Single-await linear (`AwaitValue`, `AwaitVoid`, `AwaitValueTask`) → `await` + `return`.
2. Multiple sequential awaits (`TwoSequentialAwaits`).
3. Await in branches/loops (`AwaitConditional`, `AwaitInLoop`) — transform-then-restructure.
4. Await in EH regions (`AwaitInTryFinally`).

Anything not matched falls back to honest acknowledgment (no wrong output).

**Ownership:** one agent, end to end — this is one pass family. Validate with the `--library-report` counter above and (once an overlay round-trips) a mode-aware FidelityGate follow-up.
